### ISSUE TITLE
Make `runMonitoring` take more types

### ISSUE THREAD

**Issue by mesmith75** [ganga-devs/ganga#2380]
Currently `runMonitoring` only takes job slices. Would be nice to have it accept an `int`, or a list of `int` or a `job` object.

**Comment by SreeSharvesh** [comment]
Hi! I would like to solve this issue....Can you assign it to me??

**Comment by egede** [comment]
> Hi! I would like to solve this issue....Can you assign it to me??

You are welcome to submit a pull request. Please create a fork, fix the issue and update [TestMonitoring.py](https://github.com/ganga-devs/ganga/blob/develop/ganga/GangaCore/test/GPI/Monitoring/TestMonitoring.py) to include tests that demonstrates that it is working.

**Comment by saadkhi** [comment]
Hi @egede, if this issue is still open I would to work on it.

**Comment by egede** [comment]
Yes, you are welcome. See my comment above on how to do this.

**Comment by saadkhi** [comment]
I can't able to understand what is "runMonitoring", can you briefly tell where I can find it.

**Comment by egede** [comment]
Usually in Ganga, the progress of jobs is monitored in a background thread. However, Ganga can be started without this thread running and then the `runMonitoring` function defined in the global name space can be used to monitor a specific set of jobs. See [definition](https://github.com/ganga-devs/ganga/blob/91e7df82df4148fbc9cd5cff06de5b11448bb70b/ganga/GangaCore/Core/MonitoringComponent/Local_GangaMC_Service.py#L709) and the [addition to the global name space](https://github.com/ganga-devs/ganga/blob/91e7df82df4148fbc9cd5cff06de5b11448bb70b/ganga/GangaCore/Core/__init__.py#L58). For a general overview of what jobs are and how Ganga works, see the manual.

**Comment by Sunnykumar926** [comment]
Hi @egede, this issue is still open, I would like to resolve it

**Comment by egede** [comment]
> Hi @egede, this issue is still open, I would like to resolve it

No, this one is actually currently being worked on (see #2390)

**Comment by KisuZoid** [comment]
hey @egede, I discovered this repository through the "Ganga" GSOC 25 link. If this issue is still open, I would like to work on it.

**Comment by egede** [comment]
You are welcome to work on this one, but please carefully read through the closed PR #2390 to avoid going down the same wrong route.

**Comment by KisuZoid** [comment]
@egede, Thanks for your reply and suggestion. I will figure that out and rectify the issue as soon as possible.

**Comment by saadkhi** [comment]
Hi @egede can i work on it, as i work before but at that time there is some issue in the code so issue was get closed. now it is open, so i want to contribute. Can i make a PR of my old work?

**Comment by egede** [comment]
> Hi [@egede](https://github.com/egede) can i work on it, as i work before but at that time there is some issue in the code so issue was get closed. now it is open, so i want to contribute. Can i make a PR of my old work?

Feel free to communicate with @KisuZoid about solving the issue.

**Comment by KisuZoid** [comment]
@egede  While working on runMonitoring, I noticed that on lines 1280 and 1281 in Local_GangaMC_Service.py, "logger" is not defined.

Should I create a new issue to fix this "logger", or am I misunderstanding the intended behavior of the logger function?

**Comment by egede** [comment]
> [@egede](https://github.com/egede) While working on runMonitoring, I noticed that on lines 1280 and 1281 in Local_GangaMC_Service.py, "logger" is not defined.
> 
> Should I create a new issue to fix this "logger", or am I misunderstanding the intended behaviour of the logger function?

Yes, that is for sure a bug. It is pretty hard to imagine how that line of code would be reached, but if it did, it would result in a messy "an Exception occurred within an Exception" message. Please just create a separate PR for this where you fix it.

**Comment by KisuZoid** [comment]
> > [@egede](https://github.com/egede) While working on runMonitoring, I noticed that on lines 1280 and 1281 in Local_GangaMC_Service.py, "logger" is not defined.
> > Should I create a new issue to fix this "logger", or am I misunderstanding the intended behaviour of the logger function?
> 
> Yes, that is for sure a bug. It is pretty hard to imagine how that line of code would be reached, but if it did, it would result in a messy "an Exception occurred within an Exception" message. Please just create a separate PR for this where you fix it.

I have created a separate PR that resolves the "logger is not defined" bug.

**Comment by AasmaGupta** [comment]
hey! if the issue is still open and available to work on id like to contibute.
